File: September Challenge/Week 2/Java/Day_14/HouseRobber.java
// Problem : https://leetcode.com/problems/house-robber/
// Solution:

class Solution {
    public int rob(int[] nums) {
        if(nums.length==0) return 0;
        int[] dp = new int[nums.length+1];
        dp[0]=0;
        dp[1]=nums[0];
        for(int i=1;i<nums.length;i++){
            dp[i+1]=Math.max(dp[i],nums[i]+dp[i-1]);
        }
        return dp[nums.length];
    }
}

File: September Challenge/Week 2/Cpp/Day_14/HouseRobber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int sum1 = 0,sum2 = 0;
        for(int i=0;i<nums.size();i++){
            if(!(i&1))sum1=max(sum1+nums[i],sum2);
            else sum2 = max(sum2+nums[i],sum1);
        }
        return max(sum1,sum2);
    }
};
